src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -162,8 +162,42 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
                     arg = PSObject.ToStringParser(context, ParserOps.Current(null, list));
                 }
 
+#if UNIX
+                Collection<string> resolvedPaths = new Collection<string>();
+#endif
                 if (!String.IsNullOrEmpty(arg))
                 {
+                    // This implements globbing of arguments given to native tools on UNIX.
+                    // Conversely, Windows tools implement their own globbing,
+                    // and so this should not take place.
+#if UNIX
+                    // Perform globbing if verbatim marker wasn't specified,
+                    // and there is globbing to be performed.
+
+                    // Note that there is no way to escape globbing characters,
+                    // and so no way to mix arguments with and without globbing.
+                    // It's all or nothing.
+                    if (!sawVerbatimArgumentMarker
```

Comment:
While I agree escaping is the better way to go, I think that not only can it be done separately, it really needs to be done separately as it's a much *much* larger change (and would affect many more systems, as this is the same globbing system used by cmdlets).